Extensions/messaging_tweaks.js

```diff
@@ -147,6 +152,32 @@ XKit.extensions.messaging_tweaks = new Object({
 				$(this).addClass("xkit-others_messaging_icon");
 				$(this).parents(".conversation-message").addClass("xkit-others_messaging_message");
 			}
+			if (XKit.extensions.messaging_tweaks.preferences.allow_emojis.value) {
+				var msg_div = $(this).parents(".conversation-message").find(".message-bubble .message");
+				
+				// Use regex to find emoji patterns in the chat message
+				// Find the string between the two ":" to find which image to use
+				// but only if there are emoji tags found
+				var emojis;
+				if (emojis = msg_div.text().match(/:(.*?):/gi)) {
+					for (var i = 0; i < emojis.length; i++) {
+						var emoji_tag = emojis[i].toLowerCase().substring(1).substring(0, emojis[i].length - 2);
```

Comment:
This could be more clearly written as `emojis[i].toLowerCase().substring(1, emojis[i].length - 1)` or `emojis[i].toLowerCase().substr(1, emojis[i].length - 2)`